src/monitors/rocketMonitor.cpp

```diff
@@ -0,0 +1,175 @@
+
+
+        // need to look for faces at specified fieldBoundary then find cells bordering those faces which are in specified region
+        for (PetscInt face = fStart; face < fEnd; ++face) { // Iterate through all faces to check if in fieldBoundary
+            if (ablate::domain::Region::InRegion(monitor->fieldBoundary, dm, face)) { // Check if each face is in fieldBoundary
+                PetscFVFaceGeom* fg;
+                DMPlexPointLocalRead(faceDM, face, faceGeomArray, &fg) >> checkError; // read face geometry for face
+
+                PetscInt numberNeighborCells;
+                const PetscInt* neighborCells;
+                DMPlexGetSupportSize(dm, face, &numberNeighborCells) >> ablate::checkError;
+                DMPlexGetSupport(dm, face, &neighborCells) >> ablate::checkError;
+                for (PetscInt n = 0; n < numberNeighborCells; n++) {
+
+                    // Make sure that we are not working with a ghost cell
+                    PetscInt ghost = -1;
+                    if (ghostLabel) {
+                        DMLabelGetValue(ghostLabel, neighborCells[n], &ghost);
+                    }
+                    if (ghost >= 0) {
+                        continue;
+                    }
+
+                    if (ablate::domain::Region::InRegion(monitor->region, dm, neighborCells[n])) {  // check if cell is in region
+
+                        DMPlexPointLocalRead(solDM, neighborCells[n], solArray, &conservedValues);  // Retrieve conserved values from cell
+                        monitor->computePressure = monitor->eos->GetThermodynamicFunction(eos::ThermodynamicProperty::Pressure, monitor->GetSolver()->GetSubDomain().GetFields()); // get decode state function/context
+                        monitor->computePressure.function(conservedValues, &cellPressure, monitor->computePressure.context.get()); // Retrieve pressure from cell
+                        DMPlexPointLocalFieldRead(solDM, neighborCells[n], fieldEuler.id, solArray, &cellEuler);  // retrieve euler field for density, density*velocity
+
+                        for (PetscInt d = 0; d < dim; d++) {
+                            mDotCell[d] = fg->normal[d]*cellEuler[finiteVolume::CompressibleFlowFields::RHOU+d]; // calculate mass flow rate for the cell
+                            mDotTotal[d] = mDotTotal[d] + mDotCell[d]; // summation of total mass flow rate along fieldBoundary
+                            if (abs(mDotCell[d]) > tol) {
+                                thrustCell[d] = (mDotCell[d])*((cellEuler[finiteVolume::CompressibleFlowFields::RHOU+d])/(cellEuler[finiteVolume::CompressibleFlowFields::RHO])) + (fg->normal[d])*(cellPressure-101325); // calculate thrust for the cell
+                            } else {
+                                thrustCell[d] = (fg->normal[d])*(cellPressure-101325); // calculate thrust for the cell
```

Comment:
Would you mind generalizing the reference pressure to something that could be set in the input file?